replicator/src/main/scala/com/evolutiongaming/kafka/journal/replicator/Batch.scala

```diff
@@ -98,6 +109,15 @@ private[journal] object Batch {
                     val delete = deleteOf(a.to, b.origin orElse a.origin, b.version orElse a.version)
                     delete :: tail
                   }
+
+                  val cleanTail: List[Batch] = tail.map {
+                    case appends: Appends => dropDeleted(appends, a)
+                    case delete: Delete   => delete.some
```

Comment:
I wonder if we really need previous deletes. :thinking:

Reply:
I do not see point in keeping and applying them, but it would be very challenging to remove them in current code - I've reworked it in #687